graphdatascience/query_runner/neo4j_query_runner.py

```diff
@@ -77,7 +77,29 @@ def run_query(
                 message=r"^pandas support is experimental and might be changed or removed in future versions$",
             )
 
-            return result.to_df()
+            df = result.to_df()
+
+            # Forward Neo4j warnings
+            # (see https://neo4j.com/docs/status-codes/current/notifications/#notification-categories)
+            notifications = result.consume().notifications
+
+            if notifications:
+                for notification in notifications:
+                    severity = notification["severity"]
+                    if severity == "WARNING":
+                        # FIXME handle the field deprecations
+                        if "query used a deprecated field from a procedure" in notification["description"]:
+                            continue
+
+                        if "deprecated" in notification["description"]:
+                            warning = DeprecationWarning(notification["description"])
+                        else:
+                            warning = RuntimeWarning(notification["description"])
+                        warnings.warn(warning)
```

Comment:
reading up on logging. `warnings.warn` should be used when users can fix it, while `logging.warn`should only be used if the user cannot change anything about the warning. (https://docs.python.org/3/howto/logging.html)